src/plugins/intel_gpu/src/graph/graph_optimizer/handle_reshape.cpp

```diff
@@ -201,22 +202,22 @@ void handle_reshape::run(program& p) {
                 }
             }
 
-            auto reshape_layout = node->get_output_layout();
-            auto target_format = format::get_default_format(reshape_layout.get_rank());
             auto input_layout = input_node.get_output_layout();
-            auto target_layout = layout({reshape_layout.get_partial_shape(), reshape_layout.data_type, target_format});
+            auto reshape_layout = node->get_output_layout();
+            auto preferred_output_format = format::get_default_format(reshape_layout.get_rank());
+            auto preferred_output_layout = layout({reshape_layout.get_partial_shape(), reshape_layout.data_type, preferred_output_format});
 
             if (!(node->is_output()) &&
-                 (((reshape_layout.format != target_format) && !reshape_layout.compatible(target_layout)) ||
+                 (((reshape_layout.format != preferred_output_format) && !reshape_layout.compatible(preferred_output_layout)) ||
                  (reshape_layout.format != input_layout.format))) {
                 // when some primitive does an implicit reorder to some other format then we lose the info about pitches
                 // in reshape stage we assume user provides the input vector in bfyx
-
                 // Check whether input reorder is required for format change
                 if (!format::is_simple_data_format(input_layout.format)) {
+                    auto preferred_input_format = format::get_default_format(input_layout.get_rank());
                     auto reshape_input = std::make_shared<reorder>("reorder:_reshape_input_" + node->id(),
                                                                 cldnn::input_info(input_node.id(), input_port),
-                                                                target_format,
+                                                                preferred_input_format,
```

Comment:
I have found Vladimir's comment from existing unit-tests : 
    // converts tensor to default format with rank = reshape_out_rank
    // Likely in the future we'll update that reorder so it will use reshape_input_rank
    // After that expected in format will be bfzyx